docs/proposals/job_controller.md

```diff
@@ -0,0 +1,226 @@
+
+    Create a standalone pod (not linked to any existing replication controllers)
+    Execute a new process within a given container from a running pod.
+
+**Motivation**
+
+The main goal is to provide a new pod controller type, called Job, with the following characteristics:
+
+* Be able to start of one or several pods tracked as a single entity representing a one execution of this job. Final job execution status will reflect the completion status of all pods started for this run.
+
+* Ability to trigger a new job execution either on demand (triggered through server API by an external agent or by another k8 subsystem), or periodically based on a time-based  schedule defined in job specification part (using ISO8601 syntax notation).
+
+* Be able to restart a limited number of time failing containers of pods created by one job execution (using "Restart on failure" logic extended by a maximal restart count).
+
+* Be able to limit the execution time of pods created by this job.
+
+* Be able to provide status information about pending and current job executions, along with a (limited) history of previous job runs (containing start/stop times with completion exit code)
+
+* (Future evolution) Trigger jobs based on success/failure of others.
+
+    
+**Job basic definition**
+
+The new Job object json definition for a basic implementation will have the following content:
+```
+{
+  "apiVersion": "v1beta3",
+  "kind": "Job",
+  "metadata" : {
+    "name": "myJob"
+    "labels" : {
+      "name": "myJob"
+    }
+  }
+  "spec": {
+    "schedule": {
+      "timeSpec": "R5/T01:00:00/PT01",
+      "skipOutdated": true
+      "blockOnFailure": false
+    },
+    "allowConcurrentRun": false,
+    "maxRestart": 2,
+    "execTimeout": 100,
+    "PodCount": 1,
+    "selector": { "name": "myjob"},
+    "template": {
+      "metadata" : {
+        "name": "myJobPodTemplate"
+        "labels" : {
+          "name": "myJobPodTemplate"
+        }
+      }
+      "spec": {
+        "restartPolicy": "OnFailure"
+        "containers": [{
+          "name": "job-container",
+          "image": "app/job",
+          ...
+        }]
+      }
+    }
+  }
+}
+```
+**Fields definition:**
+
+Compared to replication controller, replica count is replaced by the PodCount field (number of Pods started simultaneously for a single job execution run). 
+All other fields are characteritic of this new controller type:
+
+* "maxRestart": (default to 1) maximal number of times a failing container is allowed to be restarted.
+* "execTimeout": (not restricted by default) maximal duration a container within a pod started by the Job controller is allowed to run. Reaching this limit leads to the container to be killed.
+* "PodCount": (default to 1) number of pods started by a single run of the Job.
+* "allowConcurrentRun": (default to true) dictates if concurrent runs of this Job are allowed or not.
```

Comment:
Maybe instead of boolean value, we could have a number defining how many concurrent runs are allowed. 0 - would mean no limits, whereas number > 0 would specify that actual limit.